share/getters/getter_test.go

```diff
@@ -41,92 +43,109 @@ func TestStoreGetter(t *testing.T) {
-		randEds, namespace, dah := randomEDSWithDoubledNamespace(t, 4)
-		err = edsStore.Put(ctx, dah.Hash(), randEds)
+		randEds, namespace, eh := randomEDSWithDoubledNamespace(t, 4)
+		err = edsStore.Put(ctx, eh.DAH.Hash(), randEds)
 		require.NoError(t, err)
 
-		shares, err := sg.GetSharesByNamespace(ctx, dah, namespace)
+		shares, err := sg.GetSharesByNamespace(ctx, eh, namespace)
 		require.NoError(t, err)
-		require.NoError(t, shares.Verify(dah, namespace))
+		require.NoError(t, shares.Verify(eh.DAH, namespace))
 		assert.Len(t, shares.Flatten(), 2)
 
 		// namespace not found
 		randNamespace := sharetest.RandV0Namespace()
-		emptyShares, err := sg.GetSharesByNamespace(ctx, dah, randNamespace)
+		emptyShares, err := sg.GetSharesByNamespace(ctx, eh, randNamespace)
 		require.NoError(t, err)
 		require.Empty(t, emptyShares.Flatten())
 
 		// root not found
-		root := share.Root{}
-		_, err = sg.GetSharesByNamespace(ctx, &root, namespace)
+		emptyRoot := da.MinDataAvailabilityHeader()
+		eh.DAH = &emptyRoot
+		_, err = sg.GetSharesByNamespace(ctx, eh, namespace)
 		require.ErrorIs(t, err, share.ErrNotFound)
 	})
 
 	t.Run("GetSharesFromNamespace removes corrupted shard", func(t *testing.T) {
-		randEds, namespace, dah := randomEDSWithDoubledNamespace(t, 4)
-		err = edsStore.Put(ctx, dah.Hash(), randEds)
+		randEds, namespace, eh := randomEDSWithDoubledNamespace(t, 4)
+		err = edsStore.Put(ctx, eh.DAH.Hash(), randEds)
 		require.NoError(t, err)
 
 		// available
-		shares, err := sg.GetSharesByNamespace(ctx, dah, namespace)
+		shares, err := sg.GetSharesByNamespace(ctx, eh, namespace)
 		require.NoError(t, err)
-		require.NoError(t, shares.Verify(dah, namespace))
+		require.NoError(t, shares.Verify(eh.DAH, namespace))
 		assert.Len(t, shares.Flatten(), 2)
 
 		// 'corrupt' existing CAR by overwriting with a random EDS
-		f, err := os.OpenFile(tmpDir+"/blocks/"+dah.String(), os.O_WRONLY, 0644)
+		f, err := os.OpenFile(tmpDir+"/blocks/"+eh.DAH.String(), os.O_WRONLY, 0644)
 		require.NoError(t, err)
-		edsToOverwriteWith, dah := randomEDS(t)
+		edsToOverwriteWith, eh := randomEDS(t)
 		err = eds.WriteEDS(ctx, edsToOverwriteWith, f)
 		require.NoError(t, err)
 
-		shares, err = sg.GetSharesByNamespace(ctx, dah, namespace)
+		shares, err = sg.GetSharesByNamespace(ctx, eh, namespace)
 		require.ErrorIs(t, err, share.ErrNotFound)
 		require.Nil(t, shares)
 
 		// corruption detected, shard is removed
-		has, err := edsStore.Has(ctx, dah.Hash())
-		require.False(t, has)
-		require.NoError(t, err)
+		// try every 200ms until it passes or the context ends
+		ticker := time.NewTicker(200 * time.Millisecond)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ctx.Done():
+				t.Fatal("context ended before successful retrieval")
+			case <-ticker.C:
+				has, err := edsStore.Has(ctx, eh.DAH.Hash())
+				if err != nil {
+					t.Fatal(err)
+				}
+				if !has {
+					require.NoError(t, err)
+					return
+				}
```

Comment:
Just curious. Why do we need to wait now and didn't need to before?

Reply:
we did, but it was just flaky. passes most of the time on first try